lib/bolt/transport/orch.rb

```diff
@@ -58,7 +58,15 @@ def process_run_results(targets, results, task_name, position = [])
           # If it's finished or already has a proper error simply pass it to the
           # the result otherwise make sure an error is generated
           if state == 'finished' || (result && result['_error'])
-            if result['_error']
+            unless result['_error'].is_a?(Hash)
+              result['_error'] = { 'kind' => 'puppetlabs.tasks/task-error',
+                                   'issue_code' => 'TASK_ERROR',
+                                   'msg' => result['_error'],
+                                   'details' => {} }
+            end
+
+            result['_error']['details'] ||= {}
+            if result['_error']['details'].is_a?(Hash)
```

Comment:
We need to coerce details into a hash to match the puppet data type
https://puppet.com/docs/puppet/7.5/error-data-type.html